### DISCUSSION TITLE
Timeout issue on Windows 10 Profesional Edition.

### DISCUSSION THREAD

**Discussion by cristianprice** [hh-h/aiosnmp#48] [General]
Hello,
Thanks for this awesome library. I have an issue on windows 10. It seems that the simple example fails with aiosnmp.exceptions.SnmpTimeoutError . This seems to be an OS specific issue. On ubuntu or any other linux I cannot seem to reproduce. 
I spinned up a Wireshark and checked the packets. The communication seems fine. The problem from what I can tell comes from **aiosnmp\protocols.py** line 142. It seems that the future remains in PENDING state even though the response is received. Couldn't figure it out why. Let me know if you need any other info.
Cheers,
C

**Comment by hh-h** [comment]
hey!
1. have you tried snmpget, is it worked?
2. what python version do you use?
3. https://github.com/hh-h/aiosnmp/tree/master#source-address-host-and-port-validation have you tried this?

**Comment by cristianprice** [comment]
1. No, it seems like a linux tool. Currently I am on windows and my environment is restricted to using python stuff only. I use pysnmp without any issues so far so it is not a network issue.
2. I use Python 3.10.6 on windows 10.
3. Tried it. Doesn't seem to make any difference.

**Comment by cristianprice** [comment]
After further investigation seems to be related to this issue:
https://stackoverflow.com/questions/68502859/why-does-asyncios-datagramprotocol-stop-responding-when-an-error-is-received

**Comment by cristianprice** [comment]
Ok , confirming the bug related to the issue mentioned above. The fix would be for **Windows 10** only. The problem is that 
protocols.SnmpProtocol.datagram_received never gets called:

```
import asyncio
import aiosnmp 
import selectors

class MyPolicy(asyncio.DefaultEventLoopPolicy):
   def new_event_loop(self):
      selector = selectors.SelectSelector()
      return asyncio.SelectorEventLoop(selector)

asyncio.set_event_loop_policy(MyPolicy())

async def main():
    async with aiosnmp.Snmp(host="10.10.66.202", port=161, community="public") as snmp:
        for res in await snmp.get(".1.3.6.1.2.1.1.1.0"):
            print(res.oid, res.value)

asyncio.run(main())
```

**Comment by hh-h** [comment]
Thank you for your research! I'm not sure it is worth adding to the library itself, but I'll mention it in docs somewhere. Let's keep it open until then.

**Comment by cristianprice** [comment]
Sure. Np.
